test/integration/messaging/process-publish-message.test.js

```diff
@@ -76,6 +76,31 @@ describe('publish statement', () => {
     expect(mockSendEmail.mock.calls[0][2].personalisation.link_to_file).toBe(MOCK_PREPARED_FILE)
   })
 
+  test('should send email with scheme name', async () => {
```

Comment:
This is interesting and another valid approach to testing function calls.

I would have used the `toHaveBeenCalledWith` func and defined the params passed in that way...

There are pros and cons to both approaches which I think we'll discuss on one of our dev chats.

In terms of actions on wat's done here, I dnt think there are any. I would suggest adding a test for the Notify API key being passed to this func -- means you'll needa `jest.setup` file: https://github.com/DEFRA/ffc-pay-alerts/blob/main/jest.setup.js

Reply:
I switch between the two approaches depending on the scope of the test.

I tend to use `toHaveBeenCalledWith` for tests where I'm validating all of the args in the test, whereas here I'm only interested in the second arg and don't want to add unnecessary complexity to the test by validating the email also.  So this syntax lets me check only the argument within scope of the test.

Appreciate in this scenario validating the first email argument wouldn't really make much difference but would be a pain for scenarios with multiple arguments or where it's difficult to setup the expected argument.  If only we used a typed language, would be simpler...

Related this does let you check a specific call to the function, but that can also be achieved with `toHaveBeenNthCalledWith` but does have the same limitation of `toHaveBeenCalledWith`.

Good call with testing the API key.  Though as these tests are containerised and a fake key is already included in the docker compose test setup, no need for the Jest setup file.  Added test.